src/filters/CameraFilter.cpp

```diff
@@ -0,0 +1,33 @@
+//
+// Created by rolf on 15-12-19.
+//
+
+#include <iostream>
+#include "filters/CameraFilter.h"
+CameraFilter::CameraFilter(double observationTime, int camera) :
+        lastUpdateTime{observationTime},
+        lastMainUpdateTime{observationTime},
+        mainCamera{camera},
+        frameCount{1}{
+
+}
+int CameraFilter::frames() const {
+    return frameCount;
+}
+double CameraFilter::getLastUpdateTime() const {
+    return lastUpdateTime;
+}
+bool CameraFilter::switchCamera(int camera, double time) {
+    bool cameraSwitched = false;
+    const double TIMEOUT_TIME=0.05;
+    if (lastMainUpdateTime + TIMEOUT_TIME<time){
+        // Check if this frame is actually from a new camera
+        cameraSwitched = (camera != mainCamera);
+        mainCamera = camera;
+    }
+    if (camera == mainCamera){
+        lastMainUpdateTime=time;
+        frameCount++;//We only count frames from the main camera
+    }
+    return cameraSwitched;
+}
```

Comment:
As far as I can tell the main camera is the one who last got an update. Is this correct?

Reply:
What do you mean exactly? At what point in the code are you talking about? It can be possible that the main camera is not updated in this function. (e.g. lastMainUpdateTime + TIMEOUT_TIME>time) but camera != maincamera.